samples/ios/Continote/Continote/MyNotesViewController.swift

```diff
@@ -43,29 +43,31 @@ class MyNotesViewController: BaseViewController {
     title = Constants.Text.screenTitle
 
     // Style all buttons.
-    writeNewNoteButton.applyAppTheme()
+    newNoteButton.applyAppTheme()
   }
 
   override func viewWillDisappear(_ animated: Bool) {
     super.viewWillDisappear(animated)
 
     // Remove all bindings and database observers.
-    myNotesTableViewDataSource?.unbind()
-    myNotesTableViewDataSource = nil
-    notesFirebaseDatabaseRef?.removeAllObservers()
-    notesFirebaseDatabaseRef = nil
+    dataSource?.unbind()
+    dataSource = nil
+    notesRef?.removeAllObservers()
+    notesRef = nil
   }
 
   override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-    guard let segueIdentifier = Constants.Segue(rawValue: segue.identifier!) else { return }
+    guard let identifierString = segue.identifier,
+          let segueIdentifier = Constants.Segue(rawValue: identifierString) else { return }
 
-    switch segueIdentifier {
-    case .editNote:
+    if segueIdentifier == .editNote {
       // We need to let the EditNoteViewController know which Note the user wishes to edit
       // before the segue can be performed.
-      let cell = sender as! MyNotesTableViewCell
       let editNoteViewController = segue.destination as! EditNoteViewController
-      editNoteViewController.keyOfNoteToEdit = cell.noteKey
+      if let noteDatabaseKey:String = (sender as? MyNotesTableViewCell)?.noteKey ??
```

Comment:
Hmm... what about something like:
```
// Set the databaseKey based on the object passed in.
if let cell = sender as? MyNotesTableViewCell {
  editNoteViewController.databaseKey = cell.noteKey
} else if let key = sender as? String {
  editNoteViewController.databaseKey = key
}
```

Although it's not super pretty, I think it's a bit more clear as to what's actually going on. What do you think?